Packs/Troubleshoot/Scripts/TroubleshootExecutePlaybookByAlertQuery/TroubleshootExecutePlaybookByAlertQuery.py

```diff
@@ -0,0 +1,350 @@
+
+
+def get_playbook_info(playbook_id: str, playbook_dict: dict) -> str:
+    """
+    Retrieves the playbook information based on the provided playbook ID.
+
+    Args:
+        playbook_id (str): The playbook ID to look up.
+        playbook_dict (dict): A dictionary mapping playbook IDs to their names.
+
+    Returns:
+        str: A string containing the playbook name and ID if the ID exists; otherwise, a string with just the ID.
+    """
+    playbook_name = ""
+    if playbook_id in playbook_dict:
+        playbook_name = playbook_dict[playbook_id]
+    playbook_info = (
+        f"{playbook_name} with ID {playbook_id}" if playbook_name else f"with ID {playbook_id}"
+    )
+    return playbook_info
+
+
+def get_playbooks_dict():
+    """
+    Fetches the playbook ID-to-name mapping and stores it in a global variable.
+
+    Raises:
+        DemistoException: If the response is in an invalid format or if no playbooks are found.
+    """
+    response = demisto.executeCommand("core-api-get", {"uri": "/playbooks/idToNameMap"})
+    if not response:
+        raise DemistoException("Invalid response format while searching for playbooks.")
+
+    playbooks_dict = response[0].get("Contents", {}).get("response", {})
+    if not playbooks_dict:
+        raise DemistoException("No playbooks found. Please ensure that playbooks are available and try again.")
+
+    return playbooks_dict
+
+
+def get_playbook_id(playbook_id: str, playbook_name: str, playbooks_dict: dict):
+    """
+    Retrieve the playbook ID based on the given playbook ID or name.
+
+    Args:
+        playbook_id (str): The ID of the playbook.
+        playbook_name (str): The name of the playbook.
+
+    Raises:
+        DemistoException: If both `playbook_id` and `playbook_name` are provided,
+                          or if the playbook is not found.
+
+    Returns:
+        str: The corresponding playbook ID if found.
+    """
+    if playbook_id and playbook_name:
+        raise DemistoException("Please provide only a playbook ID or a playbook name, not both.")
+
+    if playbook_name:
+        for key_id, value_name in playbooks_dict.items():
+            if value_name == playbook_name:
+                return key_id
+
+    elif playbook_id and playbook_id in playbooks_dict:
+        return playbook_id
+
+    raise DemistoException(f"Playbook '{playbook_name or playbook_id}' wasn't found. Please check the name and try again.")
+
+
+def handle_results(command_results: dict, playbook_id: str, alert_ids: list, results_summary: RESULTS_SUMMARY):
+    """Extract and format the relevant info from the result dict.
+
+    Args:
+        command_results (dict): The results from the API call.
+        playbook_id (str): The playbook id for info.
+        alert_ids (list): A list of alert Ids for info.
+
+    Returns:
+        str: A summary of the operation status, indicating either success or the error log.
+    """
+    if not command_results:
+        return None
```

Comment:
Can the command_results return empty after the execute command. 
If so and this is nessecary "return" is sufficient.